fdbserver/RestoreLoader.actor.cpp

```diff
@@ -468,6 +461,10 @@ void _parseSerializedMutation(std::map<LoadingParam, VersionedMutationsMap>::ite
 			const uint8_t* v = vReader.consume(vLen);
 
 			MutationRef mutation((MutationRef::Type)type, KeyRef(k, kLen), KeyRef(v, vLen));
+			if (isRangeMutation(mutation)) {
+				mutation.param1 = mutation.param1 >= asset.range.begin ? mutation.param1 : asset.range.begin;
```

Comment:
Let's say the asset range is `[c, g)`. if mutation is to clear `[a, f)`, then operations here rewrite mutation to `[c, f)`. So it seems range `[a, c)` within the original mutation is lost, or is there another asset covering the range `[a, c)`?

Another scenario is that mutation is `[h, k)`, then it seems the mutation can skipped (because it is out of the asset's key range). The same is true if `mutation.param2 < asset.range.begin`.

Reply:
For the first case, there is another asset in a different version batch to cover [a, c). As described in the PR, the collection of all restore assets (or version batches) should non-overlapply cover all restore range to the target restore version.

Reply:
For the second case, you are right. The mutation can and should be skipped. I'm going to change it.